### PR TITLE
Update `extract.py` script

### LINKED ISSUE #61: Fix extract script
# Description
Currently the extract script only worked due to the black friday sale. Now that its gone, the script doesn't extract (current) price.

# Outcome
- Adapt the script so that it takes the `original_price` as the `price`(current price) if there is no discounted price shown on the website.

# User Story
As a customer, when i subscribe to a product, I want to be able to see the current price of the product (discounted or not), and be able to monitor the price trends over time on the streamlit dashboard.

### PR DESCRIPTION
**Pull request by indiah444**
- Updated `extract.py` script so that it works for all Steam links, including those behind age gates, those with promotional deals and those without. 

This closes #61

### CODE REVIEW

File: pipeline/extract.py [hunk 1]
```diff
@@ -34,8 +34,36 @@ def extract_urls_from_db() -> list[list[int, str]]:
...
+        session = requests.Session()
+        session.get(url, timeout=20)
+
+        app_id = url.split('/app/')[1].split('/')[0]
+
+        age_gate_data = {
+            "ageDay": "1",
+            "ageMonth": "January",
+            "ageYear": "1990",
+        }
```
Comment (gem09lo) [inline pipeline/extract.py:49]: added the age verification - nice!

File: pipeline/extract.py [hunk 2]
```diff
@@ -34,8 +34,36 @@ def extract_urls_from_db() -> list[list[int, str]]:
     return url_list
 
 
+def get_html_with_age_gate_bypass(url: str) -> bytes:
+    """Handles Steam URLs with age-gates by simulating form submission."""
```
Comment (gem09lo) [inline pipeline/extract.py:38]: Nice docstrings shortened!

File: pipeline/extract.py [hunk 3]
```diff
@@ -34,8 +34,36 @@ def extract_urls_from_db() -> list[list[int, str]]:
...
+    except requests.exceptions.RequestException as e:
+        logging.error("Error fetching age-gated URL %s: %s", url, e)
+        return None
+
+
 def get_html_from_url(web_page: str) -> bytes:
     """ Gets the html content from a given URL"""
+    if "store.steampowered.com" in web_page:
+        return get_html_with_age_gate_bypass(web_page)
+
```
Comment (gem09lo) [inline pipeline/extract.py:66]: Apologies comment about age was ment to be here!

File: pipeline/extract.py [hunk 4]
```diff
@@ -73,17 +101,34 @@ def scrape_from_html(html_content: bytes, url: str, product_id: int) -> dict:
...
         return None
 
-    original_price = results.find_all(
-        "div", class_="discount_original_price")[0]
-    discount_price = results.find_all(
-        "div", class_="discount_final_price")[0]
-    game_title = s.find(
+    original_price_elem = results.find(
+        "div", class_="discount_original_price")
+    discount_price_elem = results.find(
```
Comment (gem09lo) [inline pipeline/extract.py:106]: Nice spot!

File: pipeline/extract.py [hunk 5]
```diff
@@ -11,12 +11,12 @@
 """
```
Comment (gem09lo) [inline pipeline/extract.py:11]: Can you add a docstring?

**Review by gem09lo — COMMENTED** [review]
Nice fix!

File: pipeline/extract.py (hunk 5 above)
Comment (indiah444) [reply pipeline/extract.py:11]: There's a docstring at the top of the file, you have to expand it to see but it's there! <3

**Review by gem09lo — APPROVED** [review]
Nice changes! It works now!

**Pull request merged**